common/cluster/metadata.go

```diff
@@ -33,6 +33,7 @@ import (
 
 type (
 	// Metadata provides information about clusters
+	// TODO: use a more informative name, e.g. FailoverMonitor
```

Comment:
Does the TODO apply here? cluster.Metadata name seems reasonable to me for those methods.

Reply:
Yeah naming is controversial.  I just dislike the overloaded name of `metadata` everywhere. 
Maybe `ClusterMeta` will make more sense for me. 
I will remove this TODO and see if more people agree with me in the future.